pkg/controller/ground.go

```diff
@@ -1187,3 +1205,79 @@ func (op *GroundControl) podUpdate(cur, old interface{}) {
+		configContent = string(kluster.Spec.AuthenticationConfiguration)
+	} else {
+		authConfig := &apiserverv1beta1.AuthenticationConfiguration{}
+		gvks, _, err := apiserverScheme.ObjectKinds(authConfig)
+		if err != nil {
+			return fmt.Errorf("cannot determine GVK: %w", err)
+		}
+		authConfig.GetObjectKind().SetGroupVersionKind(gvks[0])
+		if swag.BoolValue(kluster.Spec.Dex) {
+			authConfig.JWT = append(authConfig.JWT, apiserverv1beta1.JWTAuthenticator{
+				Issuer: apiserverv1beta1.Issuer{
+					URL:       fmt.Sprintf("https://auth-%s.ingress.%s", kluster.Name, op.Config.Kubernikus.Domain),
+					Audiences: []string{"kubernetes"},
+				},
+				ClaimMappings: apiserverv1beta1.ClaimMappings{
+					Username: apiserverv1beta1.PrefixedClaimOrExpression{Prefix: swag.String(""), Claim: "name"},
+					Groups:   apiserverv1beta1.PrefixedClaimOrExpression{Prefix: swag.String(""), Claim: "groups"},
+				},
+			})
+		}
+		if kluster.Spec.Oidc != nil && (kluster.Spec.Oidc.IssuerURL != "" && kluster.Spec.Oidc.ClientID != "") {
+			authConfig.JWT = append(authConfig.JWT, apiserverv1beta1.JWTAuthenticator{
+				Issuer: apiserverv1beta1.Issuer{
+					URL:       kluster.Spec.Oidc.IssuerURL,
+					Audiences: []string{kluster.Spec.Oidc.ClientID},
+				},
+				ClaimMappings: apiserverv1beta1.ClaimMappings{
+					Username: apiserverv1beta1.PrefixedClaimOrExpression{Prefix: swag.String(""), Claim: "name"},
+					Groups:   apiserverv1beta1.PrefixedClaimOrExpression{Prefix: swag.String(""), Claim: "groups"},
+				},
+			})
+		}
+		var buf bytes.Buffer
+		if err := apiserverEncoder.Encode(authConfig, &buf); err != nil {
+			return fmt.Errorf("failed to encode authentication configuration: %w", err)
+		}
+		configContent = buf.String()
+
+	}
+	klusterRef := util.NewOwnerRef(kluster, v1.SchemeGroupVersion.WithKind("Kluster"))
+	cm := &api_v1.ConfigMap{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:            fmt.Sprintf("%s-auth", kluster.Name),
+			Namespace:       kluster.GetNamespace(),
+			OwnerReferences: []meta_v1.OwnerReference{*klusterRef},
+		},
+		Data: map[string]string{
+			"config.yaml": configContent,
+		},
+	}
+	existingConfigMap, err := op.Clients.Kubernetes.CoreV1().ConfigMaps(cm.GetNamespace()).Get(context.TODO(), cm.Name, meta_v1.GetOptions{})
+	if err != nil {
+		// if the configmap does not exist, create it
+		if apierrors.IsNotFound(err) {
+			_, err := op.Clients.Kubernetes.CoreV1().ConfigMaps(cm.GetNamespace()).Create(context.TODO(), cm, meta_v1.CreateOptions{})
+			return err
+		}
+		return err
+	}
+	// only update existing configmap if the content is different
+	if existingConfigMap.Data["config.yaml"] != cm.Data["config.yaml"] {
+		_, err := op.Clients.Kubernetes.CoreV1().ConfigMaps(cm.GetNamespace()).Update(context.TODO(), cm, meta_v1.UpdateOptions{})
+		op.Logger.Log(
+			"msg", "Updating authentication configuration configmap",
+			"name", cm.GetName(),
+			"kluster", kluster.GetName(),
+			"err", err)
+		return err
+	}
+	return nil
```

Comment:
With server-side apply this can potentially be shortened to a single api call.

Reply:
I tried adding that's and while its a single api call it increases the lines of code so I don't see a big benefit in changes this. During runtime the current approach will only do a single `get` configmap call unless an update is necessary. Changing this to an `server side apply` call in every reconciliation round is probably more expensive for the apiserver.